src/options.c

```diff
@@ -0,0 +1,102 @@
+#include "main.h"
+#include "options.h"
+#include "player_data.h"
+
+OPTIONS* Options_new(HeapID heap_id) {
+    OPTIONS* ret = AllocFromHeap(heap_id, sizeof(OPTIONS));
+    Options_init(ret);
+    return ret;
+}
+
+void Options_copy(OPTIONS* src, OPTIONS* dest) {
+    MI_CpuCopy8(src, dest, sizeof(OPTIONS));
+}
+
+void Options_init(OPTIONS* options) {
+    MI_CpuFill8(options, 0, sizeof(OPTIONS));
+
+    options->textSpeed = 1;   // mid speed
+    options->soundMethod = 0; // stereo
+    options->battleStyle = 0; // shift
+    options->battleScene = 0; // on
+    options->buttonMode = 0;  // normal
+    options->frame = 0;       // frame 1
```

Comment:
Might be a good idea to define enums for these. Options interface strings are [here](/pret/pokeheartgold/blob/master/files/msgdata/msg/msg_00000045.txt) and should help you define the enums.